libr/core/cmd_anal.c

```diff
@@ -2181,50 +2181,86 @@ static bool fcnNeedsPrefix(const char *name) {
 	return (!strchr (name, '.'));
 }
 
-/* TODO: move into r_anal_fcn_rename(); */
-static bool setFunctionName(RCore *core, ut64 off, const char *_name, bool prefix) {
-	char *name, *nname = NULL;
-	RAnalFunction *fcn;
-	if (!core || !_name) {
-		return false;
-	}
+static char * getFunctionName (RCore *core, ut64 off, const char *name, bool prefix) {
 	const char *fcnpfx = r_config_get (core->config, "anal.fcnprefix");
-	if (!fcnpfx) {
-		fcnpfx = "fcn";
-	}
-	if (r_reg_get (core->anal->reg, _name, -1)) {
-		name = r_str_newf ("%s.%s", fcnpfx, _name);
+	if (prefix) {
+		if (fcnNeedsPrefix (name)) {
+			if (!fcnpfx || !*fcnpfx) {
+				fcnpfx = "fcn";
+			}
+		}
 	} else {
-		name = strdup (_name);
+		fcnpfx = "";
+	}
+	if (r_reg_get (core->anal->reg, name, -1)) {
+		return r_str_newf ("%s.%08"PFMT64x, "fcn", off);
 	}
-	fcn = r_anal_get_fcn_in (core->anal, off,
-				R_ANAL_FCN_TYPE_FCN | R_ANAL_FCN_TYPE_SYM | R_ANAL_FCN_TYPE_LOC);
+	return strdup (name); // r_str_newf ("%s%s%s", fcnpfx, *fcnpfx? ".": "", name);
+}
+
+/* TODO: move into r_anal_fcn_rename(); */
+static bool setFunctionName(RCore *core, ut64 off, const char *_name, bool prefix) {
+	r_return_val_if_fail (core && _name, false);
+	char *name = getFunctionName (core, off, _name, prefix);
+	RAnalFunction *fcn = r_anal_get_fcn_in (core->anal, off,
+			R_ANAL_FCN_TYPE_FCN | R_ANAL_FCN_TYPE_SYM | R_ANAL_FCN_TYPE_LOC);
 	if (!fcn) {
 		free (name);
 		return false;
 	}
-	if (prefix && fcnNeedsPrefix (name)) {
-		nname = r_str_newf ("%s.%s", fcnpfx, name);
-	} else {
-		nname = strdup (name);
+	free (fcn->name);
+	fcn->name = name;
+	if (core->anal->cb.on_fcn_rename) {
+		core->anal->cb.on_fcn_rename (core->anal,
+				core->anal->user, fcn, name);
 	}
-	char *oname = fcn->name;
+	return true;
+}
+
+#if 0
```

Comment:
remove all this thing :/ there is git if you want to have a look at old stuff, it just makes review so much harder